nomos-utils/src/bounded_duration.rs

```diff
@@ -95,6 +97,40 @@ impl<'de, const MIN_DURATION: usize, TAG: TimeTag> DeserializeAs<'de, Duration>
     }
 }
 
+impl<const MIN_DURATION: usize, TAG: TimeTag> SerializeAs<core::time::Duration>
+    for MinimalBoundedDuration<MIN_DURATION, TAG>
+{
+    fn serialize_as<S>(source: &core::time::Duration, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let to_serialize: Self = Self {
+            duration: Duration::new(
+                source.as_secs().try_into().map_err(S::Error::custom)?,
+                source.subsec_nanos().try_into().map_err(S::Error::custom)?,
+            ),
+            _tag: PhantomData,
+        };
+        to_serialize.serialize(serializer)
+    }
+}
+impl<'de, const MIN_DURATION: usize, TAG: TimeTag> DeserializeAs<'de, core::time::Duration>
+    for MinimalBoundedDuration<MIN_DURATION, TAG>
+{
+    fn deserialize_as<D>(deserializer: D) -> Result<core::time::Duration, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let duration: Self = Self::deserialize(deserializer)?;
+        if duration.duration.is_negative() {
+            return Err(D::Error::custom(
+                "Negative duration is not supported for std::time::Duration",
+            ));
+        }
```

Comment:
This should actually never happen, as we bound the duration. But its not bad to have in case we change or expand the implementation.